### PR TITLE
virt: detect OpenStack instance

### LINKED ISSUE #23116: [OpenStack Nova]: systemd-detect-virt export none, not kvm
**Is your feature request related to a problem? Please describe.**
When the ARM system is created by OpenStack Nova, it's technically based on KVM, but `systemd-detect-virt` output is `none`, not `kvm`
> [root@kwephisprm76911 ~]$ systemd-detect-virt
> none

**Describe the solution you'd like**
`systemd-detect-virt` output `kvm` when  the system is created by OpenStack Nova

**Describe alternatives you've considered**
the product name in `/sys/class/dmi/id/product_name` is `OpenStack Nova` when the system is created by OpenStack Nova. 
> [root@kwephisprm76911 ~]$ cat /sys/class/dmi/id/product_name
> OpenStack Nova

so that when it is detected that the file `/sys/class/dmi/id/product_name` contains the words `OpenStack Nova`, systemd-detect-virt output `kvm`

**The systemd version you checked that didn't have the feature you are asking for**
main

### PR DESCRIPTION
**Pull request by YoruStar**
Fixes #23116.

### CODE REVIEW

**Comment by poettering** [comment]
hmm, how come this wasn't caught by detect_vm_cpuid()? i.e. what does it report in that CPUID VM leaf?

File: src/basic/virt.c [hunk 1]
```diff
@@ -156,6 +156,7 @@ static Virtualization detect_vm_dmi_vendor(void) {
                 Virtualization id;
         } dmi_vendor_table[] = {
                 { "KVM",                 VIRTUALIZATION_KVM       },
+                { "OpenStack Nova",      VIRTUALIZATION_KVM       },
```
Comment (poettering) [inline src/basic/virt.c]: https://cloudinit.readthedocs.io/en/latest/topics/datasources/openstack.html

Hmm, that suggests "OpenStack Compute" is also something used?

Does openstack document the used strings anywhere?

**Comment by YoruStar** [comment]
> hmm, how come this wasn't caught by detect_vm_cpuid()? i.e. what does it report in that CPUID VM leaf?

CPUID is an x86 specific interfac, but the system I use is ARM

**Comment by poettering** [comment]
ah, can you add that as a comment maybe?

File: src/basic/virt.c (hunk 1 above)
Comment (YoruStar) [reply src/basic/virt.c]: The document says that in x86 architecture , `openstack Nova` and `openstack compute` can be used to determine whether it is openstack. 
But I'm not sure if arm architecture has a product name `OpenStack Compute`

Comment (YoruStar) [reply src/basic/virt.c]: It's better that use `OpenStack` instead of `OpenStack Nova` or `OpenStack Compute`

File: src/basic/virt.c [hunk 2]
```diff
@@ -156,6 +156,7 @@ static Virtualization detect_vm_dmi_vendor(void) {
                 Virtualization id;
         } dmi_vendor_table[] = {
                 { "KVM",                 VIRTUALIZATION_KVM       },
+                { "OpenStack",           VIRTUALIZATION_KVM       }, /* Detect OpenStack instance as KVM in non x86 architectur */
```
Comment (bluca) [inline src/basic/virt.c]: typo: architectur -> architecture

Comment (YoruStar) [reply src/basic/virt.c]: fixed

**Comment by poettering** [comment]
lgtm

**Comment by poettering** [comment]
(btw, doesn't arm have some cpuid-like feature, where we can determine KVM support in a generic fashion from the CPU instead of DMI?)

**Pull request merged**